spec/xcodebuild_spec.rb

```diff
@@ -137,6 +137,51 @@ module Danger
 
       end
 
+      describe "with post message set to false" do
+
+          before do
+              data = {
+                :warnings => ['warning1', 'warning2'],
+                :ld_warnings => ['ld_warnings'],
+                :compile_warnings => ['compile_warnings'],
+                :errors => ['error1', 'error2'],
+                :compile_errors => [
+                  { :file_path => '/tmp/file1.m', :reason => 'reason1' },
+                  { :file_path => '/tmp/file2.m', :reason => 'reason2' }
+                ],
+                :file_missing_errors => [
+                  { :file_path => '/tmp/missing_file1.m', :reason => 'reason1' },
+                  { :file_path => '/tmp/missing_file2.m', :reason => 'reason2' }
+                ],
+                :undefined_symbols_errors => [{ :message => 'undefined_symbols' }],
+                :duplicate_symbols_errors => [{ :message => 'duplicate_symbols' }],
+                :tests_failures => {
+                  :suite1 => [{ :file_path => '/tmp/file1.m', :test_case => "testCase1", :reason => 'reason1' }],
+                  :suite2 => [{ :file_path => '/tmp/file2.m', :test_case => "testCase2", :reason => 'reason2' }]
+                }
+              }.to_json
+
+              filename = 'xcodebuild_tests.json'
+              allow(File).to receive(:open).with(filename, 'r').and_return(StringIO.new(data))
+
+              @xcodebuild.json_file = filename
+          end
+
+          it "must not post messages" do
+              @xcodebuild.post_message = false
+              @xcodebuild.parse_warnings
+              @xcodebuild.parse_errors
+              @xcodebuild.parse_tests
+              @xcodebuild.perfect_build
+              expect(@dangerfile.status_report[:errors]).to be_empty
+              expect(@dangerfile.status_report[:errors]).to be_empty
+              expect(@dangerfile.status_report[:warnings]).to be_empty
+              expect(@dangerfile.status_report[:messages]).to be_empty
+              expect(@dangerfile.status_report[:markdowns]).to be_empty
```

Comment:
Maybe we should add expectations for checking error and warning counts are correct?